### PR TITLE
Add inorganic-compounds batch1

### PR DESCRIPTION
**Pull request by sumitnegi7**
Have added more inorganic compounds in local-machine will push them via a new PR after this is approved.

### CODE REVIEW

**Comment by sumitnegi7** [comment]
```js
export interface SolubilitySheet {
  coldWater: Qualitative<SolubilityNature> | QuantitativeSolubility;
  hotWater: Qualitative<SolubilityNature> | QuantitativeSolubility;
  [key: string]: Qualitative<SolubilityNature> | QuantitativeSolubility;
}

```

Was not able to make coldWater and hotWater optional because of the `[key:string]` is complaining about it .

**Comment by maneetgoyal** [comment]
> ```js
> export interface SolubilitySheet {
>   coldWater: Qualitative<SolubilityNature> | QuantitativeSolubility;
>   hotWater: Qualitative<SolubilityNature> | QuantitativeSolubility;
>   [key: string]: Qualitative<SolubilityNature> | QuantitativeSolubility;
> }
> ```

export interface SolubilitySheet {
  coldWater?: Qualitative<SolubilityNature> | QuantitativeSolubility;
  hotWater?: Qualitative<SolubilityNature> | QuantitativeSolubility;
  [key: string]?: Qualitative<SolubilityNature> | QuantitativeSolubility;
}

Try this.

**Comment by sumitnegi7** [comment]
> > ```js
> > export interface SolubilitySheet {
> >   coldWater: Qualitative<SolubilityNature> | QuantitativeSolubility;
> >   hotWater: Qualitative<SolubilityNature> | QuantitativeSolubility;
> >   [key: string]: Qualitative<SolubilityNature> | QuantitativeSolubility;
> > }
> > ```
> 
> ```ts
> export interface SolubilitySheet {
>   coldWater?: Qualitative<SolubilityNature> | QuantitativeSolubility;
>   hotWater?: Qualitative<SolubilityNature> | QuantitativeSolubility;
>   [key: string]?: Qualitative<SolubilityNature> | QuantitativeSolubility;
> }
> ```
> 
> Try this.

```js
(type parameter) SolubilityNature in SolubilitySheet.Qualitative<SolubilityNature>(): any
'Qualitative', which lacks return-type annotation, implicitly has an 'any' return type.ts(7010)
'SolubilityNature' is declared but its value is never read

QuantitativeSolubility' only refers to a type, but is being used as a value here.ts(2693)

```

File: src/physical-and-chemical-data/data/inorganic-compounds.ts [hunk 1]
```diff
@@ -55,4 +55,734 @@ export const InorganicCompounds: Record<string, PhysicalProperties> = {
...
+      coldWater: {
+        nature: SolubilityNature.Decomposes,
+        adjective: Adjectives.Empty,
+        value: 'CH4',
+        temperature: {
+          value: NaN,
+          units: 'C',
+          error: 0,
+          greaterOrLess: 'less',
+        },
```
Comment (sumitnegi7) [inline src/physical-and-chemical-data/data/inorganic-compounds.ts:277]: Sir @maneetgoyal  was able to store `CH4` without modifying the interface with nature as `Decomposes` please have a look

File: src/physical-and-chemical-data/interfaces/solubility.ts [hunk 2]
```diff
@@ -4,6 +4,8 @@ export const SolubilityNature = {
   Insoluble: 'insoluble',
   Soluble: 'soluble',
   Decomposes: 'decomposes',
+  SolubleDecomposes: 'soluble decomposes',
```
Comment (maneetgoyal) [inline src/physical-and-chemical-data/interfaces/solubility.ts:7]:
```suggestion
  SolubleDecomposes: 'soluble-decomposes',
```

File: src/physical-and-chemical-data/interfaces/solubility.ts [hunk 3]
```diff
@@ -4,6 +4,8 @@ export const SolubilityNature = {
...
+  Empty: 'empty',
```
Comment (maneetgoyal) [inline src/physical-and-chemical-data/interfaces/solubility.ts:8]:
```suggestion
  Empty: '',
```

File: src/physical-and-chemical-data/interfaces/physical-properties.ts [hunk 4]
```diff
@@ -8,7 +8,7 @@ export interface PhysicalProperties {
   formula: string;
   formulaWeight: number;
   color: Qualitative<Colors>;
-  crystallineForm: Qualitative<CrystallineForms>;
+  crystallineForm: Qualitative<CrystallineForms[] | CrystallineForms>;
```
Comment (maneetgoyal) [inline src/physical-and-chemical-data/interfaces/physical-properties.ts:11]:
```suggestion
  crystallineForm: Qualitative<CrystallineForms>[] | Qualitative<CrystallineForms>;
```

File: src/physical-and-chemical-data/data/inorganic-compounds.ts [hunk 5]
```diff
@@ -55,4 +55,734 @@ export const InorganicCompounds: Record<string, PhysicalProperties> = {
...
+        adjective: Adjectives.Empty,
+      },
+      CS2: {
+        nature: SolubilityNature.Soluble,
+        adjective: Adjectives.Empty,
+      },
+    },
+  },
+
+  AluminumBromide2: {
```
Comment (maneetgoyal) [inline src/physical-and-chemical-data/data/inorganic-compounds.ts:191]:
```suggestion
  AluminumBromideHydrated: {
```

File: src/physical-and-chemical-data/data/inorganic-compounds.ts [hunk 6]
```diff
@@ -55,4 +55,734 @@ export const InorganicCompounds: Record<string, PhysicalProperties> = {
...
+        adjective: Adjectives.Empty,
+      },
+      NH4Salts: {
+        nature: SolubilityNature.Insoluble,
+        adjective: Adjectives.Empty,
+      },
+    },
+  },
+
+  AluminumBromide1: {
```
Comment (maneetgoyal) [inline src/physical-and-chemical-data/data/inorganic-compounds.ts:143]:
```suggestion
  AluminumBromide: {
```

File: src/physical-and-chemical-data/interfaces/physical-properties.ts (hunk 4 above)
Comment (sumitnegi7) [reply src/physical-and-chemical-data/interfaces/physical-properties.ts:11]: This is throwing error sir

File: src/physical-and-chemical-data/interfaces/solubility.ts [hunk 7]
```diff
@@ -27,13 +29,16 @@ export type SolubilityNature = typeof SolubilityNature[keyof typeof SolubilityNa
  * ```
  */
 export interface QuantitativeSolubility {
-  value: number;
+  value: number | string;
```
Comment (maneetgoyal) [inline src/physical-and-chemical-data/interfaces/solubility.ts:32]: Revert the change

Comment (sumitnegi7) [reply src/physical-and-chemical-data/interfaces/solubility.ts:32]: Okay sir

File: src/physical-and-chemical-data/data/inorganic-compounds.ts (hunk 1 above)
Comment (maneetgoyal) [reply src/physical-and-chemical-data/data/inorganic-compounds.ts:277]: Are you referring to `decomposes to CH4`?

**Comment by maneetgoyal** [comment]
Have pushed a couple of changes to the PR

File: src/physical-and-chemical-data/data/inorganic-compounds.ts (hunk 1 above)
Comment (sumitnegi7) [reply src/physical-and-chemical-data/data/inorganic-compounds.ts:277]: Yeah sir

Comment (maneetgoyal) [reply src/physical-and-chemical-data/data/inorganic-compounds.ts:277]: I am making some changes incrementally at the comment. Try to study the commits in parallel. If you see any issue, let me know

**Review by maneetgoyal — APPROVED** [review]

**Comment by maneetgoyal** [comment]
@sumitnegi7 Done with the changes. You can review once and merge if all looks okay.

File: src/physical-and-chemical-data/interfaces/utils.ts [hunk 8]
```diff
@@ -66,19 +67,19 @@ export type CrystallineForms = typeof CrystallineForms[keyof typeof CrystallineF
...
-  adjective: Adjectives;
+  adjective?: Adjectives;
 }
 
 export interface Measure {
   value: number;
-  units: string;
-  error: number;
-  greaterOrLess: 'greater' | 'less' | 'exact' | 'approximate';
+  units?: string;
```
[the remark this answers is not in the document]
Comment (maneetgoyal) [reply src/physical-and-chemical-data/interfaces/utils.ts:75]: Some measures can be unitless.

[the remark this answers is not in the document]
Comment (sumitnegi7) [reply src/physical-and-chemical-data/interfaces/utils.ts:75]: Okay sir Rest looks good merging now.
